### ISSUE TITLE
SklearnClassifier does not support sklearn-like estimators (e.g. ThunderSVM)

### ISSUE THREAD

**Issue by zacps** [Trusted-AI/adversarial-robustness-toolbox#836]
**Is your feature request related to a problem? Please describe.**
Scikitlearn SVMs can be slow on some datasets. GPU accelerated SVMs, such as the one in ThunderSVM, can perform better.

Currently if you pass `thundersvm.SVC` into `SklearnClassifier` an error is raised:

```python-traceback
  File "...\python\python38\lib\site-packages\art\estimators\classification\scikitlearn.py", line 76, in SklearnClassifier
    raise TypeError("Model is not an sklearn model. Received '%s'" % model.__class__)
TypeError: Model is not an sklearn model. Received '<class 'thundersvm.thundersvmScikit.SVC'>'
```

**Describe the solution you'd like**
I'd like `SklearnClassifier` to support sklearn-like classifiers.

**Describe alternatives you've considered**
Specific support for thundersvm could be added.

**Comment by zacps** [comment]
I'm not sure I'm happy with this, but the only simple solution I can think of is:

```diff
diff --git a/art/estimators/classification/scikitlearn.py b/art/estimators/classification/scikitlearn.py
index 3c94b5d7..79fa3350 100644
--- a/art/estimators/classification/scikitlearn.py
+++ b/art/estimators/classification/scikitlearn.py
@@ -1086,7 +1086,12 @@ class ScikitlearnSVC(ClassGradientsMixin, LossGradientsMixin, ScikitlearnClassif
         # pylint: disable=E0001
         import sklearn  # lgtm [py/repeated-import]

-        if not isinstance(model, sklearn.svm.SVC) and not isinstance(model, sklearn.svm.LinearSVC):
+        try:
+            from thundersvm import SvmModel as ThunderSvmModel
+        except Exception:
+            class ThunderSvmModel: pass
+
+        if not isinstance(model, sklearn.svm.SVC) and not isinstance(model, sklearn.svm.LinearSVC) and not isinstance(model, ThunderSvmModel):
             raise TypeError(
                 "Model must be of type sklearn.svm.SVC or sklearn.svm.LinearSVC. Found type {}".format(type(model))
             )
```

I think the 'right' solution would be to use the new `typing.Protocol` with `runtime_checkable` to define an interface. However, this is only available in 3.8+.

**Comment by beat-buesser** [comment]
Hi @zacps Thank you very much for sharing this idea for new features! I hadn't thought of it yet, but I think it could be useful in many applications to support scikitlearn-like libraries. In case you continue exploring `thundersvm` with ART, please let us know.
